android/src/main/java/com/brentvatne/exoplayer/PictureInPictureUtil.kt

```diff
@@ -4,48 +4,117 @@ import android.annotation.SuppressLint
 import android.app.AppOpsManager
 import android.app.PictureInPictureParams
 import android.app.RemoteAction
+import android.content.Context
+import android.content.ContextWrapper
 import android.content.pm.PackageManager
+import android.graphics.Rect
 import android.graphics.drawable.Icon
 import android.os.Build
 import android.os.Process
 import android.util.Rational
+import androidx.activity.ComponentActivity
 import androidx.annotation.ChecksSdkIntAtLeast
 import androidx.annotation.RequiresApi
 import androidx.core.app.AppOpsManagerCompat
+import androidx.core.app.PictureInPictureModeChangedInfo
+import androidx.lifecycle.Lifecycle
 import androidx.media3.exoplayer.ExoPlayer
 import com.brentvatne.common.toolbox.DebugLog
 import com.brentvatne.receiver.PictureInPictureReceiver
 import com.facebook.react.uimanager.ThemedReactContext
 
+internal fun Context.findActivity(): ComponentActivity {
+    var context = this
+    while (context is ContextWrapper) {
+        if (context is ComponentActivity) return context
+        context = context.baseContext
+    }
+    throw IllegalStateException("Picture in picture should be called in the context of an Activity")
+}
+
 object PictureInPictureUtil {
     private const val FLAG_SUPPORTS_PICTURE_IN_PICTURE = 0x400000
     private const val TAG = "PictureInPictureUtil"
 
+    @JvmStatic
+    fun addLifecycleEventListener(context: ThemedReactContext, view: ReactExoplayerView): Runnable {
+        val activity = context.findActivity()
+
+        val onPictureInPictureModeChanged: (info: PictureInPictureModeChangedInfo) -> Unit = { info: PictureInPictureModeChangedInfo ->
+            view.setIsInPictureInPicture(info.isInPictureInPictureMode)
+            if (!info.isInPictureInPictureMode && context.findActivity().lifecycle.currentState == Lifecycle.State.CREATED) {
```

Comment:
```suggestion
            if (!info.isInPictureInPictureMode && activity.lifecycle.currentState == Lifecycle.State.CREATED) {
```

How about this?

Reply:
Isn't it possible Activity is different between when bind eventListeners and when `onPictureInPictureModeChanged` is called?🤔

Reply:
It makes more sense to fix it as you said! We need to only handle Activity where the Video is visible

Reply:
[5e244e8](https://github.com/TheWidlarzGroup/react-native-video/pull/3385/commits/5e244e8f9a48f0bfdf6039f97904b5f6aa5703dc)
I've fixed it in above commit :)